### PR TITLE
images: Fixes and add services/images.go coverage.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
- Replace some function to use global service FilesService and ProducerServices.
- Replace exec.Cmd by the more testable BuildCommand.
- Fixes some minor bugs in the code.
- Add more coverage to services/images.go

FIXES: THEEDGE-3062

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [X] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by mgold1234 — APPROVED** [review]
/lgtm

File: pkg/services/images.go [hunk 1]
```diff
@@ -869,7 +870,7 @@ func (s *ImageService) addSSHKeyToKickstart(sshKey string, username string, kick
 	td := UnameSSH{sshKey, username}
 
 	s.log.WithField("templatesPath", cfg.TemplatesPath).Debug("Opening file")
-	t, err := template.ParseFiles(cfg.TemplatesPath + "templateKickstart.ks")
+	t, err := template.ParseFiles(path.Join(cfg.TemplatesPath, "templateKickstart.ks"))
```
Comment (omaciel) [inline pkg/services/images.go:873]: Love it! :)

File: pkg/services/images_test.go [hunk 2]
```diff
@@ -2121,4 +2141,493 @@ var _ = Describe("Image Service Test", func() {
...
+				// disable feature edge-management.completion_events
+				os.Unsetenv("FEATURE_COMPLETION_EVENTS")
+			})
+
+			It("should produce event models.EventTypeEdgeOstreeRepoCompleted", func() {
+				expectedRepo := models.Repo{URL: faker.URL(), Status: models.RepoStatusSuccess}
+				err := db.DB.Create(&expectedRepo).Error
+				Expect(err).ToNot(HaveOccurred())
+				mockRepoBuilder.EXPECT().ImportRepo(gomock.AssignableToTypeOf(&models.Repo{})).Return(&expectedRepo, nil)
+				mockProducerService.EXPECT().ProduceEvent(
```
Comment (omaciel) [inline pkg/services/images_test.go:2212]: Where do we assert that the event returned is the expected one? I'm sure it's my incomplete understanding of the code...

Comment (ldjebran) [reply pkg/services/images_test.go:2212]: here we are testing that ProduceEvent is called by the proper args , the two first args we know that values must be as we set it here, for the third arg , we can not get the real value but we know that it must be of type models.CRCCloudEvent. For the returned value we only assume that it should return nil , and the mock will return nil (as we are not testing Produce event, Produce event is tested in it's own module ).

@omaciel is this is what you were looking for?

Comment (ldjebran) [reply pkg/services/images_test.go:2212]: also if we will assume that error is returned from ProduceEvent,  the assertion on error should be the same as the returned from ProduceEvent, this is on the next test https://github.com/RedHatInsights/edge-api/pull/1912#discussion_r1114648932

Comment (omaciel) [reply pkg/services/images_test.go:2212]: Thanks @ldjebran for the explanation.

**Pull request merged**